Edit post and edit profile

Dmdino opened the pull request:
Добавил функционал, благодаря которому можно удалять посты и редактировать страницу профиля. Также сделал мелкие доделки, добавил refresh  в поиск и профиль пользователя.

Dmdino committed:
Edit post and edit profile

Добавил функционал благодаря которому можно удалять посты и редактировать страницу профиля. Также сделал мелкие доделки, добавил refresh  в поиск и профиль пользователя.

Antol commented on SocialMediaApp/Controller/UserProfile/EditProfileController.swift:
```diff
@@ -0,0 +1,321 @@
+import UIKit
+import Firebase
+
+class EditProfileController: UIViewController {
```
Почему вдруг решил все руками верстать? Не ошибка, просто интересно почему вдруг так делаешь

Dmdino replied:
Мне хотелось попробовать написать все руками, мне кажется, что это удобнее в плане поддержки кода и стори так тормозит, что меня просто жутко это бесит, боюсь представить, что происходит в больших проектах. Также в интернете я неоднократно натыкался на статьи и видео где говорили, что программный подход более правильный и там же нашел классные екстеншены для констрейнтов, что мне очень упростило жизнь, хотя сейчас с учетом появления SwiftUI вопрос наверно спорный. На первых лекция я задавал тебе вопрос по этому поводу и ты сказал, что более профессиональный подход делать все программно, а я доверяю твоему мнению и хочу стать профи.

Antol replied:
Отлично!) Просто обычно новички боятся, круто что решил попробовать) 
Для еще большего упрощения с работой с констрейнтами из кода могу порекомендовать вот эту либу https://github.com/SnapKit/SnapKit - надежная и проверенная 
Если захочется пойти дальше и совсем все руками лейаутить, без констрейнтов - то вот эта либа сделает это практически неотличимым от констрейнтов, но работать будет в разы быстрее https://github.com/layoutBox/PinLayout

Dmdino committed:
Camera_and_Chat

Реализовал возможность делать фото из приложения с сохранением фото в галлерею.  Добавил анимацию для боковых переходов на камеру и обратно.
